Automated Smoketest: :x:
OS: centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-166.ca-central-1.compute.internal
Sha: 5b8ba8d9a9a424688967791ae0b90961a603b445
Build: success
Build: failed!
Number of Error(s): 12
warn(226) A package without a profile could throw errors or warnings. package: dojo-util
warn(226) A package without a profile could throw errors or warnings. package: dojo-util
\terrors: 0
warn(226) A package without a profile could throw errors or warnings. package: dojo-util
warn(226) A package without a profile could throw errors or warnings. package: dojo-util
\terrors: 0
CMake Error at /home/centos/smoketest/PR-14094/build/system/aws/cmake_install.cmake:49 (file):
CMake Error at /home/centos/smoketest/PR-14094/build/system/aws/cmake_install.cmake:56 (file):
CMake Error at /home/centos/smoketest/PR-14094/build/system/aws/cmake_install.cmake:63 (file):
CMake Error at /home/centos/smoketest/PR-14094/build/system/aws/cmake_install.cmake:70 (file):
CPack Error: Error when generating package: hpccsystems-platform
make: *** [Makefile:130: package] Error 1
CMake Error at /home/centos/smoketest/PR-14094/build/system/aws/cmake_install.cmake:49 (file):
  file RPATH_CHANGE could not write new RPATH:
    /opt/HPCCSystems/lib
  to the file:
    /home/centos/smoketest/PR-14094/build/system/aws/aws-sdk-cpp/aws-cpp-sdk-core/libaws-cpp-sdk-core.so
  No valid ELF RPATH or RUNPATH entry exists in the file;
Call Stack (most recent call first):
  /home/centos/smoketest/PR-14094/build/system/cmake_install.cmake:46 (include)
  /home/centos/smoketest/PR-14094/build/cmake_install.cmake:46 (include)
CMake Error at /home/centos/smoketest/PR-14094/build/system/aws/cmake_install.cmake:56 (file):
  file RPATH_CHANGE could not write new RPATH:
    /opt/HPCCSystems/lib
  to the file:
    /home/centos/smoketest/PR-14094/build/system/aws/aws-sdk-cpp/aws-cpp-sdk-sqs/libaws-cpp-sdk-sqs.so
  The current RPATH is:
    /home/centos/smoketest/PR-14094/build/system/aws/aws-sdk-cpp/aws-cpp-sdk-core:
  which does not contain:
    /home/centos/smoketest/PR-14094/build/system/aws/aws-sdk-cpp/aws-cpp-sdk-core:/home/centos/smoketest/PR-14094/build/system/aws/aws-sdk-cpp/.deps/install/lib64:
  as was expected.
Call Stack (most recent call first):
  /home/centos/smoketest/PR-14094/build/system/cmake_install.cmake:46 (include)
  /home/centos/smoketest/PR-14094/build/cmake_install.cmake:46 (include)
CMake Error at /home/centos/smoketest/PR-14094/build/system/aws/cmake_install.cmake:63 (file):
  file RPATH_CHANGE could not write new RPATH:
    /opt/HPCCSystems/lib
  to the file:
    /home/centos/smoketest/PR-14094/build/system/aws/aws-sdk-cpp/aws-cpp-sdk-s3/libaws-cpp-sdk-s3.so
  The current RPATH is:
    /home/centos/smoketest/PR-14094/build/system/aws/aws-sdk-cpp/aws-cpp-sdk-core:
  which does not contain:
    /home/centos/smoketest/PR-14094/build/system/aws/aws-sdk-cpp/aws-cpp-sdk-core:/home/centos/smoketest/PR-14094/build/system/aws/aws-sdk-cpp/.deps/install/lib64:
  as was expected.
Call Stack (most recent call first):
  /home/centos/smoketest/PR-14094/build/system/cmake_install.cmake:46 (include)
  /home/centos/smoketest/PR-14094/build/cmake_install.cmake:46 (include)
CMake Error at /home/centos/smoketest/PR-14094/build/system/aws/cmake_install.cmake:70 (file):
  file INSTALL cannot find
  /home/centos/smoketest/PR-14094/build/system/aws/aws-sdk-cpp/.deps/install/lib64/libaws-c-common.so.1.0.0.
Call Stack (most recent call first):
  /home/centos/smoketest/PR-14094/build/system/cmake_install.cmake:46 (include)
  /home/centos/smoketest/PR-14094/build/cmake_install.cmake:46 (include)
CPack Error: Error when generating package: hpccsystems-platform
make: *** [Makefile:130: package] Error 1
Package end (2020-08-19_11-04-57 6 se
 ... ( comment is too long, 99 lines, 5693 bytes, truncated) ... 
[Build error:](https://gist.github.com/59e7e5298c60c56e82c7750be049d34a)
Build log [Build log](https://gist.github.com/HPCCSmoketest/59e7e5298c60c56e82c7750be049d34a#file-result-20-08-19-10-55-12-log)

```

--- a/.github/workflows/test-helm.yml
+++ b/.github/workflows/test-helm.yml
@@ -40,6 +40,9 @@ jobs:
         with:
           version: 'latest'
         id: install
+      - name: Install additional tools
+        working-directory: .
+        run: wget https://github.com/zegl/kube-score/releases/download/v1.8.1/kube-score_1.8.1_linux_amd64 && sudo mv kube-score_1.8.1_linux_amd64 /usr/bin/kube-score && sudo chmod +x /usr/bin/kube-score
       - name: Test
         working-directory: ./testing/helm
         run: ./run.sh

--- a/helm/hpcc/templates/_helpers.tpl
+++ b/helm/hpcc/templates/_helpers.tpl
@@ -463,16 +463,24 @@ Add any bundles
 Add security context
 Pass in a dictionary with root and me defined
 */}}
-{{- define "hpcc.addSecurityContext" -}}
-{{- if .root.Values.global.privileged }}
+{{- define "hpcc.addSecurityContext" }}
 securityContext:
+{{- if .root.Values.global.privileged }}
   privileged: true
   capabilities:
     add:
     - SYS_PTRACE
+{{- else }}
+  capabilities:
+    drop:
+    - ALL
+  allowPrivilegeEscalation: false
 {{- end }}
-{{- end -}}
-
+  runAsNonRoot: true
+  runAsUser: 999
+  runAsGroup: 1000
+  readOnlyRootFilesystem: false
+{{ end -}}
 
 {{/*
 Generate instance queue names

--- a/helm/hpcc/templates/dali.yaml
+++ b/helm/hpcc/templates/dali.yaml
@@ -14,6 +14,7 @@ spec:
       labels:
         run: {{ .name | quote }}
     spec:
+      serviceAccountName: "hpcc-default"
       initContainers: 
         {{- include "hpcc.checkDaliMount" (dict "root" $) | indent 6 }}
       containers:

--- a/helm/hpcc/templates/eclagent.yaml
+++ b/helm/hpcc/templates/eclagent.yaml
@@ -16,7 +16,7 @@ spec:
       labels:
         run: {{ .name | quote }}
     spec:
-      serviceAccountName: hpcc
+      serviceAccountName: {{ .useChildProcesses | default false | ternary "hpcc-default" "hpcc-agent" }}
       initContainers:
         {{- include "hpcc.checkDataMount" (dict "root" $) | indent 6 }}
       containers:
@@ -55,6 +55,7 @@ data:
 {{- include "hpcc.generateLoggingConfig" (dict "root" $ "me" .) | indent 6 }}
     global:
 {{ include "hpcc.generateGlobalConfigMap" $ | indent 6 }}
+{{- if not .useChildProcesses }} 
   {{ $apptype }}-jobspec.yaml: |
     apiVersion: batch/v1
     kind: Job
@@ -64,6 +65,7 @@ data:
       ttlSecondsAfterFinished: 100
       template:
         spec:
+          serviceAccountName: "hpcc-default"
           containers:
           - name: %jobname
 {{- include "hpcc.addSecurityContext" (dict "root" $ "me" .) | indent 12 }}
@@ -94,7 +96,7 @@ data:
 {{ include "hpcc.addSecretVolumes" (dict "root" $ "categories" (list "all" "ecl" "storage" ) ) | indent 10 }}
           restartPolicy: Never
       backoffLimit: 0
-      
+{{- end }}      
 ---
 {{- end }}
 {{- end }}

--- a/helm/hpcc/templates/eclccserver.yaml
+++ b/helm/hpcc/templates/eclccserver.yaml
@@ -14,7 +14,7 @@ spec:
       labels:
         run: {{ .name | quote }}
     spec:
-      serviceAccountName: hpcc
+      serviceAccountName: {{ .useChildProcesses | default false | ternary "hpcc-default" "hpcc-agent" }}
       terminationGracePeriodSeconds: {{ .terminationGracePeriodSeconds | default 600 }}
       initContainers:
         {{- include "hpcc.checkDllMount" (dict "root" $) | indent 6 }}
@@ -58,6 +58,7 @@ data:
 {{ include "hpcc.generateConfigMapQueues" $ | indent 6 }}
     global:
 {{ include "hpcc.generateGlobalConfigMap" $ | indent 6 }}
+{{- if not .useChildProcesses }} 
   compile-jobspec.yaml: |
     apiVersion: batch/v1
     kind: Job
@@ -67,6 +68,7 @@ data:
       ttlSecondsAfterFinished: 100
       template:
         spec:
+          serviceAccountName: "hpcc-default"
           containers:
           - name: %jobname
 {{- include "hpcc.addSecurityContext" (dict "root" $ "me" .) | indent 12 }}
@@ -97,7 +99,8 @@ data:
 {{ include "hpcc.addSecretVolumes" (dict "root" $ "categories" (list "all" ) ) | indent 10 }}
           restartPolicy: Never
       backoffLimit: 0
-      
+{{- end }}      
+
 ---
 {{- end }}
 {{- end }}

--- a/helm/hpcc/templates/esp.yaml
+++ b/helm/hpcc/templates/esp.yaml
@@ -15,6 +15,7 @@ spec:
       labels:
         run: {{ .name | quote }}
     spec:
+      serviceAccountName: "hpcc-default"
       containers:
       - name: {{ .name | quote }}
         workingDir: /var/lib/HPCCSystems

--- a/helm/hpcc/templates/localroxie.yaml
+++ b/helm/hpcc/templates/localroxie.yaml
@@ -20,6 +20,7 @@ spec:
         run: {{ $roxie.name | quote }}
         roxie-server: {{ $servername | quote }}
     spec:
+      serviceAccountName: "hpcc-default"
       initContainers:
         {{- include "hpcc.checkDataMount" (dict "root" $) | indent 6 }}
       containers:

--- a/helm/hpcc/templates/roxie.yaml
+++ b/helm/hpcc/templates/roxie.yaml
@@ -21,6 +21,7 @@ spec:
       labels:
         run: {{ $toponame | quote }}
     spec:
+      serviceAccountName: "hpcc-default"
       containers:
       - name: {{ $toponame | quote }}
 {{- include "hpcc.addSecurityContext" (dict "root" $ "me" .) | indent 8 }}
@@ -103,6 +104,7 @@ spec:
         run: {{ $servername | quote }}
         roxie-server: {{ $servername | quote }}
     spec:
+      serviceAccountName: "hpcc-default"
       initContainers:
         {{- include "hpcc.checkDataMount" (dict "root" $) | indent 6 }}
       containers:
@@ -161,6 +163,7 @@ spec:
         roxie-server: {{ $servername | quote }}
 {{- end }}
     spec:
+      serviceAccountName: "hpcc-default"
       initContainers:
         {{- include "hpcc.checkDataMount" (dict "root" $) | indent 6 }}
       containers:

--- a/helm/hpcc/templates/service-account.yaml
+++ b/helm/hpcc/templates/service-account.yaml
@@ -1,13 +1,19 @@
-# { {- if .Values.rbac.create -}}
+# The hpcc-default service account is used by ay component that doesn't need API access to launch child jobs
 apiVersion: v1
 kind: ServiceAccount
 metadata:
-  name: hpcc
+  name: hpcc-default
+---
+# The hpcc-agent service account is used by any component that DOES need API access to launch child jobs
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: hpcc-agent
 ---
 apiVersion: rbac.authorization.k8s.io/v1
 kind: Role
 metadata:
-  name: hpcc
+  name: hpcc-agent
 rules:
   - apiGroups: [ "" ] # core API group
     resources: [ "pods" ]
@@ -19,13 +25,12 @@ rules:
 apiVersion: rbac.authorization.k8s.io/v1
 kind: RoleBinding
 metadata:
-  name: hpcc
+  name: hpcc-agent
 roleRef:
   apiGroup: rbac.authorization.k8s.io
   kind: Role
-  name: hpcc
+  name: hpcc-agent
 subjects:
   - kind: ServiceAccount
-    name: hpcc
-    #namespace: hpcc
-#{ {- end -}}
+    name: hpcc-agent
+    namespace: {{ .Release.Namespace }}

--- a/helm/hpcc/templates/thor.yaml
+++ b/helm/hpcc/templates/thor.yaml
@@ -29,7 +29,7 @@ spec:
       labels:
         run: {{ $eclAgentName | quote }}
     spec:
-      serviceAccountName: hpcc
+      serviceAccountName: {{ $eclAgentScope.useChildProcesses | default false | ternary "hpcc-default" "hpcc-agent" }}
       containers:
       - name: {{ $eclAgentName | quote }}
         workingDir: /var/lib/HPCCSystems
@@ -66,7 +66,7 @@ spec:
       labels:
         run: {{ $thorAgentName | quote }}
     spec:
-      serviceAccountName: hpcc
+      serviceAccountName: {{ $thorAgentScope.useChildProcesses | default false | ternary "hpcc-default" "hpcc-agent" }}
       containers:
       - name: {{ $thorAgentName | quote }}
         workingDir: /var/lib/HPCCSystems
@@ -138,7 +138,7 @@ data:
       ttlSecondsAfterFinished: 100
       template:
         spec:
-          serviceAccountName: hpcc
+          serviceAccountName: {{ $thorAgentScope.useChildProcesses | default false | ternary "hpcc-default" "hpcc-agent" }}
           initContainers: 
             {{- include "hpcc.checkDataMount" (dict "root" $) | indent 10 }}
           containers:
@@ -183,7 +183,7 @@ data:
       ttlSecondsAfterFinished: 100
       template:
         spec:
-          serviceAccountName: hpcc
+          serviceAccountName: hpcc-agent
           initContainers:
             {{- include "hpcc.checkDataMount" (dict "root" $) | indent 10 }}
           containers:
@@ -221,7 +221,7 @@ data:
       ttlSecondsAfterFinished: 100
       template:
         spec:
-          serviceAccountName: hpcc
+          serviceAccountName: hpcc-default
           containers:
           - name: %jobname
 {{- include "hpcc.addSecurityContext" (dict "root" $ "me" .) | indent 12 }}

--- a/testing/helm/run.sh
+++ b/testing/helm/run.sh
@@ -41,4 +41,30 @@ do
       cat results.txt
    fi
 done
+
+if type kubeval >/dev/null 2> /dev/null; then
+   echo Running kubeval...
+   helm template $hpccchart ${options} | kubeval --strict - >results.txt 2>errors.txt
+   if [ $? -ne 0 ]
+   then
+      echo $file failed
+      cat errors.txt
+      cat results.txt
+      failed=1
+   fi
+fi
+
+if type kube-score >/dev/null 2> /dev/null; then
+   echo Running kube-score...
+   # Note we force all replicas to be > 1 as some checks are not done on replicas=1 cases e.g. antiaffinity
+   helm template $hpccchart ${options} | sed "s/replicas: 1/replicas: 2/" | kube-score score --output-format ci - >results.txt 2>errors.txt
+   if [ $? -ne 0 ]
+   then
+      echo $file failed
+      cat errors.txt
+      cat results.txt
+      failed=1
+   fi
+fi
+
 exit $failed

--- a/testing/helm/run.sh
+++ b/testing/helm/run.sh
@@ -1,6 +1,6 @@
 #!/bin/bash
 scriptdir="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
-options="--set global.image.version=latest"
+options="--set global.image.version=someversion --set global.image.pullPolicy=Always"
 hpccchart=$scriptdir/../../helm/hpcc
 failed=0
 

